lib/date_time/date_time.c

```diff
@@ -383,15 +403,39 @@ int date_time_now(int64_t *unix_time_ms)
 
 	err = date_time_uptime_to_unix_time_ms(unix_time_ms);
 	if (err) {
-		LOG_ERR("date_time_uptime_to_unix_time_ms, error: %d", err);
+		LOG_WRN("date_time_uptime_to_unix_time_ms, error: %d", err);
 		*unix_time_ms = unix_time_ms_prev;
 	}
 
 	return err;
 }
 
-int date_time_update_async(void)
+void date_time_register_handler(date_time_evt_handler_t evt_handler)
 {
+	if (evt_handler == NULL) {
+		m_evt_handler = NULL;
+
+		LOG_INF("Previously registered handler (%p) deregistered",
+			m_evt_handler);
+
+		return;
+	}
+
+	if (evt_handler) {
```

Comment:
At this point the `evt_handler` can't be `NULL` because of the check at line 415

Reply:
Removed the unnecessary check.